core-plugin/src/com/google/gct/idea/debugger/SyncResult.java

```diff
@@ -0,0 +1,80 @@
+package com.google.gct.idea.debugger;
+
+import git4idea.repo.GitRepository;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Describes the result of comparing the user's source with what the cloud debugger is
+ * attached to on the server.
+ */
+public class SyncResult {
+  private final boolean myInvalidDebuggee;
+
+  private final boolean myNeedsStash;
+  private final boolean myNeedsSync;
+  private final GitRepository myTargetLocalRepository;
+  private final boolean myHasRemoteRepository;
+  private final String myTargetSyncSHA;
+  private final String myRepoType;
+
+  SyncResult(boolean invalidDebuggee,
+             boolean needsStash,
+             boolean needsSync,
+             @Nullable String targetSyncSHA,
+             @Nullable GitRepository targetRepository,
+             boolean hasRemoteRepository,
+             String repoType) {
+    myInvalidDebuggee = invalidDebuggee;
+    myNeedsStash = needsStash;
+    myNeedsSync = needsSync;
+    myTargetSyncSHA = targetSyncSHA;
+    myTargetLocalRepository = targetRepository;
+    myHasRemoteRepository = hasRemoteRepository;
+    myRepoType = repoType;
+  }
+
+  public GitRepository getLocalRepository() {
+    return myTargetLocalRepository;
+  }
+
+  public boolean hasLocalRepository() {
+    return myTargetLocalRepository != null;
+  }
+
+  @Nullable
+  public boolean hasRemoteRepository() {
+    return myHasRemoteRepository;
+  }
+
+  @Nullable
+  public String getTargetSyncSHA() {
+    return myTargetSyncSHA;
+  }
+
+  public boolean isValidDebuggee() {
+    return !myInvalidDebuggee;
+  }
+
+  /**
+   * @return whether the local repository has uncommitted changes we need to stash
```

Comment:
let's stick to style and always have a method javadoc comment (when a comment is required), but leave the @ javadoc s optional.